syntax = "proto3";

package rkt;

service RocketteService {
  rpc UpdateIntData (IntValue) returns (StatusResponse) {}
  rpc UpdateFloatData (FloatValue) returns (StatusResponse) {}
  rpc UpdateStringData (StringValue) returns (StatusResponse) {}
  rpc UpdateBytesData (BytesValue) returns (StatusResponse) {}
}

message StatusResponse {
  int32 status = 1;
}

message StringValue {
  string identifier = 1;
  string value = 2;
}

message IntValue {
  string identifier = 1;
  int32 value = 2;
}

message FloatValue {
  string identifier = 1;
  float value = 2;
}

message BytesValue {
  string identifier = 1;
  bytes value = 2;
}